src/subdomains/generic/gs/gs.controller.ts

```diff
@@ -0,0 +1,37 @@
+import { Controller, Post, UseGuards, Body, Get, Query } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { ApiBearerAuth, ApiExcludeEndpoint } from '@nestjs/swagger';
+import { RoleGuard } from 'src/shared/auth/role.guard';
+import { UserRole } from 'src/shared/auth/user-role.enum';
+import { DbQueryBaseDto, DbQueryDto } from './dto/db-query.dto';
+import { SupportReturnData } from './dto/support-return-data.dto';
+import { GsService } from './gs.service';
+
+@Controller('gs')
+export class GsController {
+  constructor(private readonly gsService: GsService) {}
+
+  @Post('db')
+  @ApiBearerAuth()
+  @ApiExcludeEndpoint()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.SUPPORT))
+  async getRawData(@Body() query: DbQueryDto): Promise<any> {
+    return await this.gsService.getRawData(query);
+  }
+
+  @Post('db/bankTx')
+  @ApiBearerAuth()
+  @ApiExcludeEndpoint()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.SUPPORT))
+  async getBankTxExtendedData(@Body() query: DbQueryBaseDto): Promise<any> {
+    return await this.gsService.getExtendedData(query);
+  }
```

Comment:
It's a bit weird to have a dedicated endpoint, but a generic service method and DTO 😉